parser/Builder.cc

```diff
@@ -1206,6 +1206,10 @@ class Builder::Impl {
                 // Label key is a symbol `sym: val`
                 return match_var_hash(label->loc, key->val.show(gs_));
             } else if (auto *key = parser::cast_node<DSymbol>(pair->key.get())) {
+                if (key->nodes.empty()) {
```

Comment:
For what it's worth, it looks like all string-quoted keys are not allowed in the punned hash key/val pairs

```ruby
❯ ruby -e '{"x":}'
-e:1: syntax error, unexpected '}'
{"x":}
```

not just empty strings.